internal/cmd/list_bundles.go

```diff
@@ -0,0 +1,45 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	mtcli.AddCommand(listBundlesCmd)
+}
+
+var (
+	listBundlesExamples = []string{
+		"  #List all the bundles present in an index image.",
+		"  mtcli list-bundles <index_image>",
```

Comment:
It seems if we provide the `--addon-name` flag, it would simplify the code so much. Also, since this is the cache key, it should help with performance.

If I understand correctly, by providing `--addon-name` we could remove the new function altogether `ExtractAndParseAllAddons` and load manifests based on the `addonName` instead of iterating over all operators?

I think it's a reasonable ask, tenants will for sure know the addon-name when using `$ mtcli list-bundles` and for our use case, which is to validate index images before pushing, we will also know the addon-name.

WDYT?

Otherwise it seems I don't see the difference between `ExtractAndParseAddon` and the new `ExtractAndParseAllAddons`?

Reply:
@sugarraysam The issue with that approach is that indexImages can come with other dependent operators, like the ocs-addon. So it'll be very awkward to type in the names of the addon + all the dependent addons.

Reply:
Yep, makes sense.